src/trade.js

```diff
@@ -12,6 +12,15 @@ class Trade {
     this._expiresAt = null;
     this._delegate = delegate;
     this._api = api;
+    this._fromApi = false;
+
+    if (obj !== null) {
+      this._id = obj.id;
+    }
+  }
+
+  setFromAPI () {
+    this._fromApi = true;
```

Comment:
Why does it matter if the trade was set from the api or not? Doesn't seem right to have the behavior of the trade depend on where the data came from.